### DISCUSSION TITLE
Is it possible to describe @ISA in the specification?

### DISCUSSION THREAD

**Discussion by yuki-kimoto** [Ovid/Cor#74] [Q&A] [answered]
Is it possible to describe <code>@ISA</code> in the Section 4: Classes?

https://github.com/Ovid/Cor/blob/master/rfc/classes.md#422-inheritance

In the following FAQ, I heard the class specified by <code>:isa</code> is added <code>@ISA</code>. the roles specified by <code>:does</code> is not added to <code>@ISA</code>.

https://github.com/Ovid/Cor/discussions/72

Or <code>@ISA</code> is not used?

**Comment by Ovid** [comment]
`@ISA` should not be used. Corinna is single inheritance. There would be no need for an array of parent classes. Code _outside_ of Corinna should, if it really needs type information, be relying on either `if ( $object->isa('Foo') ) { ... }` or the MOP.

Even in current Perl, roles would never be added to `@ISA` because their behavior is flattened into the class. They're not parent classes.

**Reply by yuki-kimoto** [reply]
Thanks. I understood now. <code>@ISA</code> is not used in Corinna.

**Reply by yuki-kimoto** [reply]
I have another question.

If <code>@ISA</code> is not used, how does <code>$self->next::method()</code> work?

I think <code>$self->next::method()</code> searches the method from parent classes(<code>@ISA</code>) by the mro order.

**Reply by Ovid** [reply]
You would have to ask @leonerd that. He's the one implementing it.

**Reply by yuki-kimoto** [reply]
OK. I will ask @leonerd in this topic.

**Comment by leonerd** [comment, marked as answer]
I wouldn't say it doesn't use `@ISA` under the hood, at all. Just you shouldn't interact with it as a regular Perl developer.

Internally the usual inheritance mechanisms are still at work as normal.

**Reply by yuki-kimoto** [reply]
Thanks!

> Internally the usual inheritance mechanisms are still at work as normal.

OK. I wanted to know this.

**Reply by leonerd** [reply]
FYI, latest implementation of the `feature-class` branch now unit-tests that reading `@ISA` still gives the expected results, and forbids any caller from mutating the `@ISA` array of a true `class`, so as not to confuse everything.

**Reply by yuki-kimoto** [reply]
Thanks. I realize that <code>@ISA</code> mechanism to search methods in the super class is used in <code>use feature 'class'</code>.
